cmake/modules/kconfig.cmake

```diff
@@ -152,9 +152,15 @@ set(COMMON_KCONFIG_ENV_SETTINGS
   PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE}
   srctree=${ZEPHYR_BASE}
   KERNELVERSION=${KERNELVERSION}
+
+  VERSION_MAJOR=${APP_VERSION_MAJOR}
+  VERSION_MINOR=${APP_VERSION_MINOR}
+  PATCHLEVEL=${APP_PATCHLEVEL}
+  VERSION_TWEAK=${APP_VERSION_TWEAK}
```

Comment:
Ah thanks, sorry about that! Had my test app misconfigured 🤦

Reply:
My error was putting `source "Kconfig.zephyr"` at the bottom of my `<app>/Kconfig` file where I was referencing the symbols, instead of before, like this:

```kconfig
source "Kconfig.zephyr"

config APP_VERSION
    string "Application version"
    default "$(VERSION_MAJOR).$(VERSION_MINOR).$(PATCHLEVEL)"
```